### PR TITLE
New preview version for Maintenance extension

### PR DESCRIPTION
**Pull request by sotaneja**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by sotaneja** [comment]
@jsntcy can I get a review please?

File: src/maintenance/setup.py
Comment (changlong-liu) [inline src/maintenance/setup.py:10]: If we have had 1.0.1 released, I think we need a bigger version number:
https://github.com/Azure/azure-cli-extensions/pull/2214/files#diff-ba12df623efca123165ca48252645955R13
https://github.com/Azure/azure-cli-extensions/pull/2214/files#diff-2503d1af174da5a5c12c1c6542850ce9R6

File: src/maintenance/setup.py
Comment (jsntcy) [inline src/maintenance/setup.py:13]: Is it expected that the version is downgraded from 1.0.1 to 0.1.0?

File: src/maintenance/azext_maintenance/generated/commands.py
Comment (jsntcy) [inline src/maintenance/azext_maintenance/generated/commands.py:34]: the command is changed, before it is "applyupdate", but now it 's "apply-update", is this breaking change expected?

File: src/maintenance/azext_maintenance/azext_metadata.json
Comment (jsntcy) [inline src/maintenance/azext_maintenance/azext_metadata.json:2]: Is it expected that all commands labelled as "Experimental" as before it's GA?

File: src/maintenance/azext_maintenance/generated/commands.py
Comment (jsntcy) [inline src/maintenance/azext_maintenance/generated/commands.py:72]: Don't label the existing commands as experimental as they are GA before; other existing commands as well.

File: src/maintenance/setup.py
Comment (sotaneja) [reply src/maintenance/setup.py:13]: yes, I wasn't sure about this, thanks for confirming.

File: src/maintenance/azext_maintenance/generated/commands.py
Comment (jsntcy) [reply src/maintenance/azext_maintenance/generated/commands.py:72]: Please just update is_experimental label in "Generated" folder as there is no good mechanism to overwrite it.

---
In reply to: [478229448](https://github.com/Azure/azure-cli-extensions/pull/2214#discussion_r478229448) [](ancestors = 478229448)

File: src/maintenance/setup.py
Comment (jsntcy) [reply src/maintenance/setup.py:13]: You can use new version "1.1.0"

---
In reply to: [478221129](https://github.com/Azure/azure-cli-extensions/pull/2214#discussion_r478221129) [](ancestors = 478221129)

File: src/maintenance/HISTORY.rst
Comment (jsntcy) [inline src/maintenance/HISTORY.rst:5]: Two versions here:
1.0.1
1.1.0

File: src/maintenance/azext_maintenance/azext_metadata.json
Comment (jsntcy) [reply src/maintenance/azext_maintenance/azext_metadata.json:2]: Use "azext.isPreview": true here

---
In reply to: [478224362](https://github.com/Azure/azure-cli-extensions/pull/2214#discussion_r478224362) [](ancestors = 478224362)

File: src/maintenance/azext_maintenance/generated/commands.py
Comment (jsntcy) [inline src/maintenance/azext_maintenance/generated/commands.py:25]: Use is_preview=True here.

**Comment by jsntcy** [comment]
@sotaneja, could you please update the PR based on the comments above so that we can have another review before merging?

File: src/maintenance/azext_maintenance/generated/commands.py
Comment (sotaneja) [reply src/maintenance/azext_maintenance/generated/commands.py]: regenerated with existing command names

**Comment by sotaneja** [comment]
@jsntcy I have made your PR suggestions. Thanks.

File: src/maintenance/azext_maintenance/generated/_params.py
Comment (jsntcy) [inline src/maintenance/azext_maintenance/generated/_params.py:111]: No need to register "resource_group_name" as  it will be registered as global arguments automatically.

File: src/maintenance/azext_maintenance/generated/_params.py
Comment (jsntcy) [inline src/maintenance/azext_maintenance/generated/_params.py:173]: Define duplicated arguments as CLIArgumentType for reuse, please refer to other codes.

**Review by jsntcy — APPROVED** [review]
:shipit:

**Pull request merged**
